### PR TITLE
Multivip status update

### PR DESCRIPTION
**Pull request by aaha97**
Status update for Ingress and Service Type LB is updated
following cases are tested.
VIP <-> MultiVip
VIP <-> Public IP
VIP -> (MultiVIP <-> Public) -> VIP

UTs added

### CODE REVIEW

**Comment by aaha97** [comment]
build ako

**Comment by aaha97** [comment]
build ako

**Comment by monotosh-avi** [comment]
@aaha97  are you taking care of ip deletion case for multi vip ?

**Comment by aaha97** [comment]
build ako

**Comment by aaha97** [comment]
> @aaha97 are you taking care of ip deletion case for multi vip ?

yes, the scenario for multivip to single vip is tested

File: internal/rest/avi_obj_vsvip.go [hunk 1]
```diff
@@ -449,6 +450,17 @@ func (rest *RestOperations) AviVsVipCacheAdd(rest_op *utils.RestOp, vsKey avicac
 						continue
 					}
 					vsvipVips = append(vsvipVips, addr)
+					floating_ip, valid := vip["floating_ip"].(map[string]interface{})
+					if !valid {
+						utils.AviLog.Infof("key: %s, msg: invalid type for floating_ip in vsvip: %s", key, name)
```
Comment (monotosh-avi) [inline internal/rest/avi_obj_vsvip.go:455]: This should be a warn log

File: internal/rest/avi_obj_pool.go [hunk 2]
```diff
@@ -268,13 +268,24 @@ func (rest *RestOperations) AviPoolCacheAdd(rest_op *utils.RestOp, vsKey avicach
...
+				for _, vsvipkey := range vs_cache_obj.VSVipKeyCollection {
+					vsvip_cache, ok := rest.cache.VSVIPCache.AviCacheGet(vsvipkey)
+					if ok {
+						vsvip_cache_obj, found := vsvip_cache.(*avicache.AviVSVIPCache)
+						if found {
+
+							if len(vsvip_cache_obj.Fips) == 0 {
+								IPAddrs = vsvip_cache_obj.Vips
+							} else {
+								IPAddrs = vsvip_cache_obj.Fips
```
Comment (monotosh-avi) [inline internal/rest/avi_obj_pool.go:281]: Can there be a case where both vips and  fips are empty ?

File: internal/rest/avi_obj_vs.go [hunk 3]
```diff
@@ -378,14 +378,25 @@ func (rest *RestOperations) StatusUpdate(rest_op *utils.RestOp, vs_cache_obj *av
 					utils.AviLog.Infof("key: %s, msg: found pool: %s, will update status", key, poolkey.Name)
 					pool_cache_obj, found := pool_cache.(*avicache.AviPoolCache)
 					if found {
-						IPAddr := vs_cache_obj.Vip
-						if vs_cache_obj.Fip != "" {
-							IPAddr = vs_cache_obj.Fip
+						var IPAddrs []string
```
Comment (monotosh-avi) [inline internal/rest/avi_obj_vs.go:381]: This block of code to obtain IPAddrs is being executed multiple times, can we add this to a common function and reuse the code ?

**Review by monotosh-avi — COMMENTED** [review]
Thanks @aaha97 for getting this done. Looks good to me. A couple of minor comments.

File: internal/rest/avi_obj_pool.go (hunk 2 above)
Comment (aaha97) [reply internal/rest/avi_obj_pool.go:281]: no, vips should be present at least

File: internal/rest/avi_obj_vsvip.go (hunk 1 above)
Comment (aaha97) [reply internal/rest/avi_obj_vsvip.go]: updated

**Pull request merged**